TEST_CHECKLIST_GUIDELINES.md

```diff
@@ -0,0 +1,56 @@
+Testing Scripts should
+- Be easy to follow
+- Be grouped in a logical way (either page or overarching component, like "Login" or "Filters" or "Saved search") as much as possible
+- Reference other tests or groups of tests (ie: when we are done testing saved searches, we make sure that collections are also OK since we might have impacted them) 
+- Assume a basic comprehension of the system, to not be overly verbose. 
+
+    ie: we can assume that the user knows how to login if that's a feature we have had for a long time, OR at least link to the login test checklist in the "Login" instruction if we need to be more specific 
+
+- Explicitly test for non happy paths
+
+ie:  If you are desgining a login test checklist, it should include validation errors
+
+- Reflect problems we had in the past (while doing dev or testing, etc), to keep adding to our test suites as we move forward with the platform development.  
+    
+    Example: if you faced a bug like "the filters don't close when I click the x button" make sure you create a checklist for that. 
+
+- Have tests that consist of steps and preconditions
+
+    Examples: 
+    - [Test Script] Saved searches and filters
+    - [TEST GROUP] The "SAVE SEARCH" button is displayed correctly
+    - [TEST Title] The SAVE SEARCH button is displayed when there are filters.
+        - Precondition: There are no filters applied.
```

Comment:
This feels very manual, how do we enumerate all of them? Would it make sense to have an example of the most atomic level for one of these tests? Or may be to have some kind of "state" concept?

Reply:
yea, it is very manual. :/ that's kind of the nature of a checklist tho. Unless we roll our own or buy some super fancy testing tool (not sure which of them has this concept but I agree it would be nice) I don't see a way around it. 
Also, keep in mind that the whole thing is manual in this doc.
There has to be a separate section on e2e testing, which IMO is completely necessary at this point and should be a priority. Having this manual tests is a stepping point for then translating them to tests in our suite. But I still think you absolutely need the english only suite, because you need non- developers to check things too, you need it as documentation for the company,  and because it forces you to write it down and think through everything. I admit that to me, the process of writing the checklist is probably as valuable as the end result.

Something else about e2e is that right now we lack proper classes and ids in a bunch of parts of the app, and end of doing selectors like  "second span after tag <bla> with sibling X" instead of  "first span with this class inside this id" when doing e2e testing (at least for me that was the experience when I did it a couple weeks ago while testing testcafe) . That's another "to solve" thing for sure..

Reply:
> you need non- developers to check things too
clarifying: this is mostly because you want to have access to good documentation on the system behavior.You need a good reference that you can share with Jess for instance and say : "this is how we thought this would work when we wrote our tests, is it working correctly? can you edit it and improve it? "  then we can adjust our suite to match that.
But, after writing it, I also think "yea but that should have been in the ticket and in the manuals instead, and the test suite would have gotten the info from the manual and story ideally". So maybe we need to sit down and think more about this in general because it might point to a bigger issue in our process. :/